python/src/gen_gherkin/generate_gherkin_feature_files_impl.py

```diff
@@ -4,43 +4,273 @@
+    status = ExecutionStatus.SUCCESS
+
+    return new_files, ExecutionResult(plugin_name, "gen-gherkin-behaviors", status, messages)
+
+
+def collect_models(parsed_models: list[dict]) -> list:
+    """
+    Return a structured dict like parsed_models, but only consisting of model definitions.
+
+    Args:
+        parsed_models (list(dict)): A list of parsed definitions
+
+    Returns:
+        A list containing only model definitions
+    """
+    collected_models = []
+    for model in parsed_models:
+        if model.get_root_key() == "model":
+            collected_models.append(model)
+
+    return collected_models
+
+
+def sanitize_scenario_step_entry(step: str) -> str:
+    """
+    Remove any conflicting keyword from the scenario step.
+
+    Args:
+        step (str): A scenario step
+
+    Returns:
+        The scenario step with conflicting keywords removed
+    """
+    if does_step_start_with_gherkin_keyword(step):
+        return step.split(None, 1)[1]
+    return step
+
+
+def collect_and_sanitize_scenario_steps(scenario: dict) -> list[dict]:
+    """
+    Collect and sanitize scenario steps then return template properties for a 'scenarios' entry.
+
+    Args:
+        scenario (dict): The scenario definition from a model
+
+    Returns:
+        A list of template properties
+    """
+    if "description" in scenario:
+        description = scenario["descrption"]
+    else:
+        description = "TODO: Fill out this feature description."  # noqa: T101
+    scenario_steps = [
+        {
+            "definition": description,
+            "givens": [sanitize_scenario_step_entry(given) for given in scenario["given"]],
+            "whens": [sanitize_scenario_step_entry(when) for when in scenario["when"]],
+            "thens": [sanitize_scenario_step_entry(then) for then in scenario["then"]],
+        }
+    ]
+    return scenario_steps
+
+
+def collect_behavior_entry_properties(behavior_entry: dict) -> list[dict]:
+    """
+    Produce a list of template property dictionaries from a behavior entry.
+
+    Args:
+        behavior_entry (dict): The behavior definition from a model
+
+    Returns:
+        A list of template property dictionaries.
+    """
+    feature_name = behavior_entry["name"]
+    if "description" in behavior_entry:
+        feature_description = behavior_entry["description"]
+    else:
+        feature_description = "TODO: Fill out this feature description."  # noqa: T101
+    behavior_requirements = []
+    scenario_lists = []
+    if "acceptance" in behavior_entry:
+        for acceptance in behavior_entry["acceptance"]:
+            if "scenarios" in acceptance:
+                for scenario in acceptance["scenarios"]:
+                    scenario_lists.append(collect_and_sanitize_scenario_steps(scenario))
+    if "requirements" in behavior_entry:
+        for requirement in behavior_entry["requirements"]:
+            behavior_requirements.append(requirement)
```

Comment:
So, not certain these if checks really need to be here since there is no special handling for when the check is false. So, could just have the for loops. Those fields should be caught by the constraint evaluating definition structure of allowable fields and let the user know if they've provided something unexpected, so not sure there is a point in having the else return that notice and therefore not certain what is gained by having the if loops.

Reply:
I believe it is possible for a definition to not have those fields, and in that case it will just skip that part and continue.

Reply:
Correct that the definition is not currently required to have the fields. And correct then the logic would be skipped. 

However, I think it then brings up a point for discussion. If a behavior does not have an `acceptance` field, which is where the actual behavior is defined, not sure the behavior entry makes much sense to include? So, maybe we need to mark `acceptance` as a required aspect of the structure. 

The `requirements` aspect I can understand not wanting to make a required entry because someone could just be sketching something that is not derived from requirements. So, maintaining the if check to allow for that portion to be skipped makes sense.

Reply:
I removed the if checks, but I am now thinking that I should put them back.  All models are being put through this function, and it could be possible for one model in a system to have a behavior and another to not.  So if a system has multiple models it would need to check them.